DependencyInjection/Compiler/TweakCompilerPass.php

```diff
@@ -40,7 +40,7 @@ public function process(ContainerBuilder $container)
             // Replace empty block id with service id
             if (strlen($arguments[0]) == 0) {
                 $definition->replaceArgument(0, $id);
-            } elseif ($id != $arguments[0]) {
+            } elseif ($id != $arguments[0] && 0 !== strpos('Sonata\\BlockBundle\\Block\\Service\\', $definition->getClass())) {
```

Comment:
using the class rather than the service id. people are good at copy-pasting and might just use sonata.block.service for their own services where we want the deprecation notice